src/stirling/source_connectors/socket_tracer/uprobe_manager.cc

```diff
@@ -296,24 +296,13 @@ static constexpr const auto kLibSSLMatchers = MakeArray<SSLLibMatcher>({
         .libcrypto = "libcrypto.so.3",
         .search_type = HostPathForPIDPathSearchType::kSearchTypeEndsWith,
     },
-#ifdef PL_BPF_TESTING
-    // TODO(ddelnano): Remove this once Pixie can TLS trace libpython linked interpreters.
-    //
-    // This ifdef allows the test for TLS tracing python 3.10 and later to succeed. Python 3.10
-    // and later require probing the _ex variants of SSL_write/SSL_read, but depending on the
-    // python interpreter may also require probing an so file not probed today
-    // (libpython${VERSION}.so).
-    //
-    // Our BPF test uses a distroless python container image, which ships with an interpreter built
-    // with --enable-shared. This results in a libpython shared library that contains openssl
-    // and other dynamically linked library's symbols. Please see
-    // https://github.com/pixie-io/pixie/issues/1347 for more background and the follow up work.
     SSLLibMatcher{
-        .libssl = "libpython3.10.so.1.0",
-        .libcrypto = "libpython3.10.so.1.0",
-        .search_type = HostPathForPIDPathSearchType::kSearchTypeEndsWith,
+        // This must match independent of python version and INSTSONAME suffix
+        // (e.g. libpython3.10.so.0.1).
+        .libssl    = "libpython",
+        .libcrypto = "libpython",
+        .search_type = HostPathForPIDPathSearchType::kSearchTypeContains,
```

Comment:
I opted to keep this search type simple to avoid a more expensive regex match (with a `std::string` allocation). The potential risk is that this matches more than what we can successful uprobe, but I think the likelihood of that is low. In the event it did match something, the uprobe attachment would fail and Stirling would move on to the next binary.

Since this logic is triggered for each library path in `/proc/$PID/maps` for every pid that is probed, I felt this was the more conservative choice.